modules/drivers/mongo/test/metabase/driver/mongo_test.clj

```diff
@@ -220,6 +220,89 @@
                        :database-position 0}}}
            (driver/describe-table :mongo (mt/db) (t2/select-one Table :id (mt/id :venues)))))))
 
+(deftest sync-indexes-info-test
+  (mt/test-driver :mongo
+    (mt/dataset (mt/dataset-definition "composite-index"
+                  ["singly-index"
+                   [{:field-name "indexed" :indexed? true :base-type :type/Integer}
+                    {:field-name "not-indexed" :indexed? false :base-type :type/Integer}]
+                   [[1 2]]]
+                  ["compound-index"
+                   [{:field-name "first" :indexed? false :base-type :type/Integer}
+                    {:field-name "second" :indexed? false :base-type :type/Integer}]
+                   [[1 2]]]
+                  ["multi-key-index"
+                   [{:field-name "url" :indexed? false :base-type :type/Integer}]
+                   [[{:small "http://example.com/small.jpg" :large "http://example.com/large.jpg"}]]])
+
+      (try
+       (testing "singly index"
+         (is (true? (t2/select-one-fn :database_indexed :model/Field (mt/id :singly-index :indexed))))
+         (is (false? (t2/select-one-fn :database_indexed :model/Field (mt/id :singly-index :not-indexed)))))
+
+       (testing "compount index"
+         (mongo.util/with-mongo-connection [conn (mt/db)]
+           (mcoll/create-index conn "compound-index" {"first" 1 "second" 1}))
```

Comment:
To be explicit that this test relies on the ordering of fields in the index map, I'd suggest using `array-map` here as you do in the next test